models/points_test.go

```diff
@@ -1772,3 +1772,24 @@ t159,label=another a=2i,value=1i 1`
 		t.Fatalf("expected 2 points, got %d", len(points))
 	}
 }
+
+func TestNewPointsWithBytesWithCorruptData(t *testing.T) {
+	ch := make(chan struct{})
+	go func() {
+		corrupted := []byte{0, 0, 0, 3, 102, 111, 111, 0, 0, 0, 4, 61, 34, 65, 34, 1, 0, 0, 0, 14, 206, 86, 119, 24, 32, 72, 233, 168, 2, 148}
+		p, _ := models.NewPointFromBytes(corrupted)
+		p.Fields()
+		close(ch)
+	}()
+	select {
+	case _ = <-ch:
+	case _ = <-time.NewTimer(time.Second).C:
+		t.Fatalf("failed: probable infite loop")
+	}
+}
+
+func TestNewPointsRejectsEmptyFieldNames(t *testing.T) {
+	if _, err := models.NewPoint("foo", nil, models.Fields{"": 1}, time.Now()); err == nil {
+		t.Fatalf("expected error if field with empty field name is created - empty string")
```

Comment:
nit: the failure message could be something like `got nil, expected error`, i.e., it's preferable to write what you got and then what you expected.